DeratControl.Domain/Entities/Facility.cs

```diff
@@ -12,10 +14,24 @@ public class Facility : EntityBase<int>
         public virtual Organization Organization { get; private set; }
         public virtual ICollection<Perimeter> Perimeters { get; private set; }
         public virtual ICollection<Review> Reviews { get; private set; }
+
         private Facility()
         {
             this.Perimeters = new List<Perimeter>();
```

Comment:
I'd recommend initializing collections inline, not in constructor, because having other constructors you can miss this initialization so your collections could potentially be null